prymer/primer3/primer3_input.py

```diff
@@ -77,26 +90,52 @@
 PRIMER_WT_SIZE_GT -> 0.1
 PRIMER_WT_TM_LT -> 1.0
 PRIMER_WT_TM_GT -> 1.0
+PRIMER_WT_SELF_ANY_TH -> 0.0
+PRIMER_WT_SELF_END_TH -> 0.0
+PRIMER_WT_HAIRPIN_TH -> 0.0
 """
 
+from dataclasses import MISSING
 from dataclasses import dataclass
+from dataclasses import fields
 from typing import Any
+from typing import Optional
 
 from prymer.api.span import Span
 from prymer.primer3.primer3_input_tag import Primer3InputTag
-from prymer.primer3.primer3_parameters import Primer3Parameters
+from prymer.primer3.primer3_parameters import PrimerAndAmpliconParameters
+from prymer.primer3.primer3_parameters import ProbeParameters
 from prymer.primer3.primer3_task import Primer3TaskType
-from prymer.primer3.primer3_weights import Primer3Weights
+from prymer.primer3.primer3_weights import PrimerAndAmpliconWeights
+from prymer.primer3.primer3_weights import ProbeWeights
 
 
 @dataclass(frozen=True, init=True, slots=True)
 class Primer3Input:
-    """Assembles necessary inputs for Primer3 to orchestrate primer and/or primer pair design."""
+    """Assembles necessary inputs for Primer3 to orchestrate primer, primer pair, and/or internal
+    probe design."""
 
     target: Span
     task: Primer3TaskType
-    params: Primer3Parameters
-    weights: Primer3Weights = Primer3Weights()
+    primer_and_amplicon_params: Optional[PrimerAndAmpliconParameters] = None
+    probe_params: Optional[ProbeParameters] = None
+    primer_weights: Optional[PrimerAndAmpliconWeights] = None
+    probe_weights: Optional[ProbeWeights] = None
```

Comment:
Is there a reason not to allow default weights?  I.e. does primer3 blow up if you set probe weights and then design left primers?

Reply:
Yes.

Reply:
I find your answer confusing.  I'm asking "if you provide too many weights, isn't that ok?", but the last line of your answer - "it doesn't know how far off optimal it is when optimal is left undefined." - makes it sound like you're answering "what if we didn't provide enough weights"?

Reply:
To use your example: Primer3 blows up if you set probe weights and then design left primers. By giving `ProbeWeights()`, you are telling Primer3 how you want the resultant designs to be scored. But Primer3 complains because you didn't give it `ProbeParameters()` to score against -- that is what I meant by "optimal is undefined". 

Primer3 does not know what to do with the weights you provide it _in the absence of corresponding parameters_. Because each set of parameters is now defaulted to `None`, that is the context we are working within. 

Specifically, it is the `PRIMER_INTERNAL_WT_GC_PERCENT_*` weights that cause the problem, here is the error when you specify `PRIMER_INTERNAL_WT_GC_PERCENT_LT` and `PRIMER_INTERNAL_WT_GC_PERCENT_GT` but ask for left primers only:
```
PRIMER_ERROR=Hyb probe GC content is part of objective function while optimum gc_content is not defined
=
primer3_core: Hyb probe GC content is part of objective function while optimum gc_content is not defined
```

Reply:
Circling back to this with one more thought.  Would it be hard to still have these defaulted on the input, but then depending on the task, only translate the appropriate weights to tags and send them to primer3?